### DISCUSSION TITLE
Setup Lora E19-433M0S to work at full power at 1W

### DISCUSSION THREAD

**Discussion by saviour84** [jgromes/RadioLib#1354] [Q&A]
Dear helpers

Trust this questions finds you well and in good health in these festive days,

I encountered difficulty trying to transmit and receive characters from more than 300 m apart by two transceiver modules bought from Sparkfun (LoRa 1W Breakout, E19-433M30S) with a matched antenna at 50 ohms impedance. The codes for transmitter and receiver can be found at the end of the post. When the power command (lora.setOutputPower(30)) was set at 30 dbm (1W), the Arduino library didn’t accept it and returned an error (RadioLib.h). Moreover, when set at 20 dBm, the transmission worked but with a weak signal (high RSSI). Both receiver and transmitter codes of the Sparkfun modules can be seen at the end for your kind attention.

The strange thing is that when I bought cheaper and less powerful modules (10mW DRFZ1278F), they were able to transmit more than 1 km.

I read other guests post comments; they said: to set the power at 4 dBm on the library (RadioLib.h) so the PA gain is 26 dBm to make 30 dBm in total. I did it, but the outcome was a weaker signal (high RSSI) that shortened the communication distance.

As suggested by the user manual, I supplied the module with 5V/1A. 

I also found on your RadioLib.h startup settings on this link: https://github.com/jgromes/RadioLib/blob/master/examples/SX127x/SX127x_Settings/SX127x_Settings.ino.

Might the initial startup configurations be wrong?

  
Kindly could you provide a tentative code snippet or otherwise for the startup of the modules so I could be able to increase much more the distance of transmission? Or how to enable the module to work at 1W output transmission? 

Modules and antenna links

https://www.sparkfun.com/products/22792
https://www.sparkfun.com/products/20856

```c++
#include <RadioLib.h>

// SX1276 has the following connections: NSS, DIO0, DIO1, RST, MOSI, MISO, SCK
SX1276 lora = new Module(10, 2, 9, 3); // NSS, DIO0, RESET, DIO1

int txPower = 20; //

void setup() {
Serial.begin(9600);

// Initialize LoRa communication at the desired frequency (e.g., 433 MHz or 915 MHz)
Serial.print(F("Initializing LoRa… "));
int state = lora.begin(433.0);
if (state == RADIOLIB_ERR_NONE) {
Serial.println(F(“success!”));
} else {
Serial.print(F("failed, code "));
Serial.println(state);
while (true);
}

// Set output power
lora.setOutputPower(txPower);

Serial.print(F(“Transmit power set to: “));
Serial.print(txPower);
Serial.println(F(” dBm”));
}

void loop() {
// Send packet every second
String message = “LoRa Test Packet”;
Serial.print(F("Sending packet: "));
Serial.println(message);

int state = lora.transmit(message);

// Check for transmit success
if (state == RADIOLIB_ERR_NONE) {
Serial.println(F(“success!”));
} else {
Serial.print(F("failed, code "));
Serial.println(state);
}

// Wait a second before sending another packet
delay(1000);
```

Receiver Code

```c++
#include <RadioLib.h>
#include <math.h>

// SX1276 module configuration
SX1276 lora = new Module(10, 2, 9, 3); // NSS, DIO0, RESET, DIO1

// Propagation parameters
float A = -71; // RSSI at 1 meter (calibrated value)
float n = 2.0; // Path loss exponent (adjust based on environment)

void setup() {
Serial.begin(9600);

// Initialize LoRa communication at the desired frequency (433 MHz for this example)
Serial.print(F("Initializing LoRa… "));
int state = lora.begin(433.0);
if (state == RADIOLIB_ERR_NONE) {
Serial.println(F(“success!”));
} else {
Serial.print(F("failed, code "));
Serial.println(state);
while (true);
}

Serial.println(F(“LoRa Receiver Ready”));
}

void loop() {
// Receive LoRa packet
String str;
int state = lora.receive(str);

// Check if a packet was received
if (state == RADIOLIB_ERR_NONE) {
// Packet received successfully
Serial.print(F("Received packet: "));
Serial.println(str);

// Get RSSI
int rssi = lora.getRSSI();
Serial.print(F("RSSI: "));
Serial.println(rssi);

// Calculate and display estimated distance
float distance = calculateDistance(rssi, A, n);
Serial.print(F("Estimated distance: "));
Serial.print(distance);
Serial.println(F(" meters"));
} else if (state == RADIOLIB_ERR_RX_TIMEOUT) {
// Timeout error, no packet received
[//Serial.println](https://serial.println/)(F(“Receive timeout”));
} else {
// Some other error occurred
Serial.print(F("Receive failed, code "));
Serial.println(state);
}

// Wait a moment before checking for another packet
delay(1000);
}

// Function to calculate distance based on RSSI
float calculateDistance(int rssi, float A, float n) {
// RSSI distance formula: d = 10^((A - RSSI) / (10 * n))
float distance = pow(10, (A - rssi) / (10 * n));
return distance;
}
```

Thank you very much

Saviour

**Comment by jgromes** [comment]
> When the power command (lora.setOutputPower(30)) was set at 30 dbm (1W), the Arduino library didn’t accept it and returned an error

I believe [this wiki](https://github.com/jgromes/RadioLib/wiki/High-power-Radio-Modules-Guide) page will explain a lot.

> but the outcome was a weaker signal (high RSSI)

What do you mean by high? Your RSSI is probably negative, so by high do you mean "more negative number"? For example, -100 dBm is less than -90 dBm.

> to set the power at 4 dBm on the library (RadioLib.h) so the PA gain is 26 dBm to make 30 dBm in total.

That depends on the manufacturer of your module. There are many "high power" radio modules out there with various PAs.

In your code, you are also missing RF switch control. The module you have linked to has RXEN and TXEN pins. See the example Sparkfun provided on their website.

**Comment by saviour84** [comment]
I appreciate your help,

For now I will ignore the said Lora Module (E19-433M30S), because neither the manufacturer (Ebyte) nor Sparkfun give support on these modules. For now, I will focus on the DRFZ1278 modules because they give me better results, although they have less powerful transmission. I will keep you updated from now on the results using the other module.

All the best for these festive days,

Saviour
